### ISSUE TITLE
Create a basic tutorial (Improve documentation)

### ISSUE THREAD

**Issue by wezz** [maxdee/alc_freeliner#44]
It took me a while to even figure out exactly how to install and load everything.

Now I've got freeliner up and running but due to all the keyboard shortcuts it's quite hard to figure out how to make any progress in making a visual.

It would greatly help to have either a written step by step guide or a screencast with someone instructing on how to make your first visuals. A Hello World for freeliner so to say.

I think it looks like an awesome piece of software and I would really like to try it out if I could only get over this first bump in the road

**Comment by maxdee** [comment]
Check the [how to freeline](https://github.com/maxdee/alc_freeliner#how-to-freeline), most people can figure it out from that. Dont forget to go to `http://localhost:8000/index.html` for the web gui once freeliner is running.

**Comment by wezz** [comment]
I've succeeded in:
Installing Processing and the related libraries.
Gotten the freeliner up and running including the Web GUI.

But even when I follow the guide to the letter I don't seem to get any animation.
I've been messing around for a while now, creating shapes and trying to animate them but without success. It feel like I'm either missing a piece of the puzzle, or something is broken.

![image](https://user-images.githubusercontent.com/1185533/42418106-cf2a0e88-8299-11e8-8f16-9e387097ddaa.png)

Making any kinds of changes in the Web GUI or trying to use the command keys for animating gives no effect.
A more detailed "your first animation" tutorial would be greatly appreciated. I'm usually not a fan of video tutorials but in this case I thing it would be really useful

**Comment by maxdee** [comment]
Ok so when you make a shape do you hit `shift + A` or any other letter like
`shift + B`? This will add a template (animation parameters) to the shape.
Let me know if it works.

**Comment by Jacques-Olivier-Farcy** [comment]
I put a pull request for a step by step basic tutorial

**Comment by louprgb** [comment]
@wezz Did you manage to find how to create animation?

**Comment by wezz** [comment]
I got some basic animations going but due to the lack of documentation / guides I never got the natural hang of it.  Haven't touched freeliner in a long time now.

Thought about creating my own web based alternative instead but never had the time to create it.

**Comment by maxdee** [comment]
I have been fixing stuff in freeliner regularly, some recent updates will bring us to 0.5 and I will stop changing things once and for all. I added project directories and better saving / loading, but documentation has not been updated yet. Its good to know people are interested, it will motivate me to update the docs a bit. Its lots of work.

**Comment by louprgb** [comment]
Ok thanks for your replies, but could you help me to solve my problem? How
can i animate my geometries? I have crated a geometry, placed a center with
"c". But when i push "a" or "b" the animation works "half the time" for the
first geometry which has been assigned automatically to the template "M".
But when i create a second geometry, the software doesn't animate this one
because he doesn't place into a template.

I followed all your instructions in your website, and tried all the
shortcuts, without success.

So,
what is the correct procedure in order to animate one object? And also If I
want to create several in its own template?

Thank you

**Comment by Jacques-Olivier-Farcy** [comment]
Hi @louprgb , did you have a look at :
https://github.com/maxdee/alc_freeliner/blob/devel/doc/step-by-step.md

**Comment by louprgb** [comment]
Yes, as i said in my previous email, I followed every step by step
tutorial. But when I click "a" to generate a render it doesn't work
everytime for the first geometry and never for the others. And i cannot
manage to choose which template yo assign to each geometry.

**Comment by Jacques-Olivier-Farcy** [comment]
I am also intersted to continue using this tool, and maybe if we are some people, we could help to make a more detail documentation ...
I am in France, we can use some visio / stream stuff so i can show you how i use it even if i am far from an expert on freeliner

**Comment by louprgb** [comment]
I'm french but living in Montreal.
Très bonne idée pour le visio. Quand êtes vous disponible ?

**Comment by Jacques-Olivier-Farcy** [comment]
Je suis assez dispo lundi et mardi prochain après-midi.
Et un aperçu de ce que j'ai testé :
https://interstices.ouvaton.org/mapping.html

**Comment by devinenicolas** [comment]
@Jacques-Olivier-Farcy Sympa comme utilisation. 
Salut Max !

**Comment by maxdee** [comment]
Si ça vous tente de faire une session live stream de tuto/tests/doc je suis partant. Je veux vraiment compléter le release 0.5, la version ultime en Processing, j'y suis presque! Je pourrais dédier lundi et mardi pour faire un sprint.

**Comment by Jacques-Olivier-Farcy** [comment]
Je suis partant aussi.
Je réserve mon lundi / mardi après-midi au moins, ce qui sera les mâtinées pour vous, j'aurai peut-être du temps pour poursuivre dans la soirée de mon côté.
Tu pensais à quoi comme plateforme pour se retrouver ?

**Comment by louprgb** [comment]
Lundi et mardi ca me va à partir de 9h (heure de montreal) 15h en France.
Ca te vas?

Tu veux faire ça par zoom?

**Comment by maxdee** [comment]
Lets use google hangouts, or jitsi if people prefer that.

**Comment by louprgb** [comment]
Jitsi if its ok for you. Will you create the group for the video chat?

**Comment by Jacques-Olivier-Farcy** [comment]
Oui zoom ça me va, et  l'horaire aussi.
J'essaierai de faire aussi du stream via twitch ou autre en parallèle pour que vous puissiez avoir une vue d'ensemble.

**Comment by Jacques-Olivier-Farcy** [comment]
I also prefer Jitsi (Privacy and open-source)

**Comment by louprgb** [comment]
Pas de problème pour Jitsi. Envoi moi ton invit une fois que t'es prêt.

J'ai un compte Twitch aussi (@teddykitano) tu peux m'ajouter également
dessus.

**Comment by maxdee** [comment]
rendez-vous sur https://gitlab.com/maxd/alc_freeliner/-/issues/5 pour continuer la disscussion!
